### PR TITLE
Begin writing contributing guidelines

### LINKED ISSUE #21: Documentation should talk about how to get started with the project

### PR DESCRIPTION
**Pull request by harrykeightley**
Closes #21

### CODE REVIEW

**Review by nicklambourne — COMMENTED** [review]
What's here looks good so far, I'll hold off on a proper review until you're done (ping me).

**Comment by harrykeightley** [comment]
Issues and feature requests are not yet included but this is good enough to be useful for people wanting to contribute to documentation or code.

Also interested to know if @nicklambourne thinks this proposed contributing flow is best practice.

File: docs/contributing.md [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
+- About: about.md
+...
+- Your Page Title: your_page.md
+```
+
+Your should now see your page in the sidebar!
+
+### `mkdocs` development commands
+
+- `mkdocs new [dir-name]` - Create a new project.
```
Comment (nicklambourne) [inline docs/contributing.md:39]: Is this particular command relevant, I assume we've done this initial setp?

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+And you're ready to go!
+
+#### Cloud run Services
+
+To prepare the services devlopment environment:
+
+- [Make sure you have poetry installed](https://python-poetry.org/docs/)
+- Optionally, setup the appropriate python version with a tool like `asdf` or `pyenv`.
+  Currently we're using python 3.10 for our python functions and services.
```
Comment (nicklambourne) [inline docs/contributing.md:108]:
```suggestion
  Currently we're using Python 3.10 for our Python functions and services.
```

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+`poetry install`
+
+And you're ready to go!
+
+#### Cloud run Services
+
+To prepare the services devlopment environment:
+
+- [Make sure you have poetry installed](https://python-poetry.org/docs/)
+- Optionally, setup the appropriate python version with a tool like `asdf` or `pyenv`.
```
Comment (nicklambourne) [inline docs/contributing.md:107]:
```suggestion
- Optionally, setup the appropriate Python version with a tool like `asdf` or `pyenv`.
```

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+Navigate to the relevant service folder:
+
+`cd services/trainer`
+
+or
+
+`cd services/transcriber`
+
+And install the poetry dependencies:
```
Comment (nicklambourne) [inline docs/contributing.md:119]:
```suggestion
And install the Poetry dependencies:
```

File: docs/contributing.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+### Commit your changes
+
+When you're happy with the code on your computer, you need to commit your changes:
+
+`git commit -a`
+
+This should fire up your editor to write a commit message. When you have finished,
+save, and close to continue.
+
+**TODO** Commit format
```
Comment (nicklambourne) [inline docs/contributing.md:220]: Again, would suggest leaving these out, or at least being consistent in how you format the TODOs.

File: docs/contributing.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+git rebase my_new_branch
+git checkout my_new_branch
+```
+
+If there are no conflicts, and after running the tests, all seems well at the inn,
+its time to proceed.
+
+### Forking (general contributors)
+
+First, navigate to the [elpiscloud github repo](https://github.com/CoEDL/elpiscloud)
```
Comment (nicklambourne) [inline docs/contributing.md:244]: Might be worth explaining briefly why forking is necessary for non-org contributors.

Comment (nicklambourne) [reply docs/contributing.md:244]: Or leading with the CoEDL section.

File: docs/contributing.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+
+```
+git rebase my_new_branch
+git checkout my_new_branch
+```
+
+If there are no conflicts, and after running the tests, all seems well at the inn,
+its time to proceed.
+
+### Forking (general contributors)
```
Comment (nicklambourne) [inline docs/contributing.md:242]:
```suggestion
### Forking (Public Contributors)
```

File: docs/contributing.md [hunk 8]
```diff
@@ -0,0 +1,304 @@
+# Contributing to Elpis Cloud
```
Comment (nicklambourne) [inline docs/contributing.md:1]: From a readibility standpoint may be worth adopting Title Case throughout the headings in this file.

File: docs/contributing.md [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+
+Before you start the process of submitting a PR, it is important to write some
+tests for your feature (if applicable).
+
+For our backend, we're using `pytest` and `pytest-mock` for testing.
+On the frontend, we don't currently have tests, but this may change in the future.
+
+#### Testing poetry environments
+
+From within a poetry folder (`functions` or `services/**`), you can run the
```
Comment (nicklambourne) [inline docs/contributing.md:176]:
```suggestion
From within a Poetry folder (`functions` or `services/**`), you can run the
```

File: docs/contributing.md [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+To run the non-integration tests from within a service, you can instead run:
+
+`poetry run pytest -m 'not integration'`
+
+Which will skip those tests.
+
+#### Writing your own tests
+
+When writing your own tests, I'd recommend following the style of the preexisting
+tests for that folder. Generally in poetry folders, the tests exist under the
```
Comment (nicklambourne) [inline docs/contributing.md:191]:
```suggestion
tests for that folder. Generally in Poetry folders, the tests exist under the
```

File: docs/contributing.md [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+```
+
+Now apply those changes to your branch:
+
+```
+git rebase my_new_branch
+git checkout my_new_branch
+```
+
+If there are no conflicts, and after running the tests, all seems well at the inn,
```
Comment (nicklambourne) [inline docs/contributing.md:239]: While I personally don't mind folksy documentation, I've been told that colloquialisms like this don't translate well for ESL contributors.

File: docs/contributing.md (hunk 6 above)
Comment (nicklambourne) [inline docs/contributing.md:244]:
```suggestion
First, navigate to the [elpiscloud GitHub repo](https://github.com/CoEDL/elpiscloud)
```

File: docs/contributing.md [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+
+If changes are requested, push to your forked feature branch, and those changes
+should now be visible on the PR.
+
+When addressing feedback comments in a PR, if your latest commit fixes a comment
+from a previous revision, mark it as resolved to keep the PR clean.
+
+While you're waiting for feedback on your pull request, open up a few other
+pull requests and give someone else some! They'll appreciate it in the same
+way that you appreciate feedback on your patches.
```
Comment (nicklambourne) [inline docs/contributing.md:304]:
```suggestion
While you're waiting for feedback on your pull request, open up another
pull request, or give someone else feedback on theirs! They'll appreciate 
it in the same way that you appreciate feedback on your patches.
```

**Review by nicklambourne — CHANGES_REQUESTED** [review]
A few suggestions here and there, but looking great.

File: docs/contributing.md [hunk 13]
```diff
@@ -1,13 +1,48 @@
 # Contributing to Elpis Cloud
 
-Say hi
-
 ## Reporting an Issue
 
 ## Making feature requests
```
[the remark this answers is not in the document]
Comment (harrykeightley) [reply docs/contributing.md:5]: Fixed in latest revision.

File: docs/contributing.md (hunk 1 above)
Comment (harrykeightley) [reply docs/contributing.md:39]: Removed in next revision.

File: docs/contributing.md (hunk 2 above)
Comment (harrykeightley) [reply docs/contributing.md:108]: Fixed in next revision.

File: docs/contributing.md (hunk 3 above)
Comment (harrykeightley) [reply docs/contributing.md:107]: Fixed in next revision.

File: docs/contributing.md (hunk 4 above)
Comment (harrykeightley) [reply docs/contributing.md:119]: Fixed in next revision.

File: docs/contributing.md (hunk 5 above)
Comment (harrykeightley) [reply docs/contributing.md:220]: Fixed in latest revision.

File: docs/contributing.md (hunk 6 above)
Comment (harrykeightley) [reply docs/contributing.md:244]: Fixed in latest revision.

File: docs/contributing.md (hunk 7 above)
Comment (harrykeightley) [reply docs/contributing.md:242]: Fixed in latest revision.

File: docs/contributing.md (hunk 8 above)
Comment (harrykeightley) [reply docs/contributing.md:1]: Fixed in latest revision.

File: docs/contributing.md (hunk 9 above)
Comment (harrykeightley) [reply docs/contributing.md:176]: Fixed in latest revision.

File: docs/contributing.md (hunk 10 above)
Comment (harrykeightley) [reply docs/contributing.md:191]: Fixed in latest revision.

File: docs/contributing.md (hunk 11 above)
Comment (harrykeightley) [reply docs/contributing.md:239]: Hmmm which inn? Good point. Fixed in latest revision.

File: docs/contributing.md (hunk 6 above)
Comment (harrykeightley) [reply docs/contributing.md:244]: Fixed in latest revision.

File: docs/contributing.md [hunk 14]
```diff
@@ -0,0 +1,321 @@
...
+<!---
+## Reporting an Issue
+
+## Making feature requests
+
+-->
+
+## Contributing to the Documentation
+
+To help flesh out our documentation at `docs.elpis.com`, you'll first need to install
```
Comment (nicklambourne) [inline docs/contributing.md:12]:
```suggestion
To help flesh out our documentation at [docs.elpis.com](https://docs.elpis.com), you'll first need to install
```

**Review by nicklambourne — APPROVED** [review]
LGTM 👍

**Pull request merged**